src/routes/todoRoutes/index.ts

```diff
@@ -0,0 +1,37 @@
+import { Router } from 'express'
+
+const todoRoutes = Router()
+const BASE_PATH = '/todo'
```

Comment:
Essa variável faz sentido existir aqui? Pergunto pois se essa rota nunca mudar, não teria necessidade de criar uma const para isso, agora se caso exista essa possibilidade não seria melhor colocar nas variáveis de ambiente?

O que você acha?

Reply:
A rota provavelmente nunca ira mudar. O intuito de uma const aqui eh de garantir um pouco mais de escabilidade e manutenabilidade, isto eh, quando outro desenvolvedor tiver que adicionar uma nova rota, essa variavel garantiria que pelo menos o BasePath seria igual aos demais. Alem disso, caso o time achasse que o nome da rota ficou ruim, bastava atualizar uma variavel que refletiria em todas as rotas.
Gostaria de reforcar tbm que nessa aplicacao a rota em questao eh bem simples e curta, mas imagine em uma aplicacao que teria uma rota composta por mais pastas `'admin/gerenciar/dashboard/detalhes' `e nesse `detail` teria ao menos 5 verbos para serem criados e aplicados na rota, em um caso como esse, a chance do desenvolvedor errar a escrita eh muito grande, o que levaria a um possivel erro (404 not found).
Faz sentido?

Reply:
Faz sentido sim, nesse caso então entraria no segundo ponto que mencionei, não seria melhor então centralizar essas consts nas variáveis de ambiente (.env) para se ter um ponto único para se alterar, pois se houvesse uma alteração em todas as rotas, eu não precisaria entrar arquivo por arquivo de routes para alterar cada um.

Reply:
Acredito que o proposito da variavel de ambiente nao se aplica aqui. No meu entendimento variaveis de ambiente sao utilizadas para proteger dados sensiveis e quando temos uma configuracao que nao sera necessariamente a mesma quando executada em outro computador. (Exemplo: a porta, o banco essas configs devem ser setadas no ambiente que for ser selecionado para rodar). Portanto nesse caso as rotas sempre devem ser as mesma em todos os ambientes, de forma que se um alterar, isso deve ser replicado em todos os computadores que executarem essa aplicacao.
Um possivel engano que poderia ocorrer caso fosse uma variavel de ambiente seria o seguinte:
Imagine que alguem erre o nome da rota ao escreve la na hora de subir o ambiente e ao inves de user o cara escreva usuario, dessa forma a tabela do banco ira se chamar user quando a rota que ira atualizar a tabela user se chama usuario. O ponto eh que a informacao fica descasada, fiz um exemplo tosco de traducao, mas poderiamos extrapolar para algo bem pior. Faz sentido?

Reply:
Faz sentido sim. Boa explicação!